### ISSUE TITLE
Issues with current version

### ISSUE THREAD

**Issue by RandomGuy26** [UNINETT/SAML-tracer#33]
Since the latest version was released, it doesn't appear to be working with any version of FireFox that I've found. One workaround that had work was to use FireFox ESR, but then that too quit working.

Anyone seeing this problem? Is there a workaround?

**Comment by khlr** [comment]
Hi!

Which Firefox versions did you try?
Do you use Windows or Linux?

**Comment by RandomGuy26** [comment]
Windows 7

Current version I tried was ESR 52.5.3, tried the version before it as well. Also used 32-bit FireFox 56.0 and SAML 0.4.0 but letting it update to 1 breaks it.

Sent from my iPhone

On Jan 18, 2018, at 1:40 PM, Jan Köhler <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi!

Which Firefox versions did you try?
Do you use Windows or Linux?

**Comment by Bigfoot0007** [comment]
52.5.3 (32-bit), can 't work now.

**Comment by RandomGuy26** [comment]
Right I have tried MANY versions of FireFox I can't recall them all.

Is there a version of Firefox where SAML Tracer 1 works with? Maybe that's a better approach to the issue.

Sent from my iPhone

On Jan 19, 2018, at 7:36 AM, Lyndon <notifications@github.com<mailto:notifications@github.com>> wrote:


52.5.3 (32-bit), can 't work now.

**Comment by thijskh** [comment]
I've just reïnstalled SAML Tracer 1.0 from addons.mozilla.org from scratch on FF 57.0.4 on Linux and that just works. Only thing is the known (FF) bug that the window is initially blank unless you right-click it.

Perhaps it helps if you describe what doesn't work exactly or how it fails.

**Comment by RandomGuy26** [comment]
FF ESR 52.5.3 64-bit the right clicking on the window did not work, I will try to find a 57 version.

Sent from my iPhone

On Jan 19, 2018, at 8:26 AM, Thijs Kinkhorst <notifications@github.com<mailto:notifications@github.com>> wrote:


I've just reïnstalled SAML Tracer 1.0 from addons.mozilla.org<http://addons.mozilla.org> from scratch on FF 57.0.4 on Linux and that just works. Only thing is the known (FF) bug that the window is initially blank unless you right-click it.

Perhaps it helps if you describe what doesn't work exactly or how it fails.

**Comment by khlr** [comment]
First of all: Any SAML Tracer version before v1.0 won't work from FF57 onwards, since the support for XPCOM/XUL-based extensions [was dropped with that FF-version](https://blog.mozilla.org/addons/2016/11/23/add-ons-in-2017/).

Consequently, with v1.0 SAML Tracer has been converted from a XUL-based extension to a WebExtension. This type of extension will be compatible from FF57 onwards, but _may_ not be compatible with older FF-versions (depending on whether the respective FF-version already implemented the follwing JavaScript APIs or not):

SAML Tracer v1.0+ uses the following [JavaScript APIs](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs):

* browserAction
  * onClicked 45
* extension
  * getURL 45
* webRequest
  * onBeforeRequest 46
  * onBeforeSendHeaders 45
  * onHeadersReceived 45
* windows
  * create 45
  * update 45

As you can see, most APIs are available since FF45, `onBeforeRequest` came with FF46. So _in theory_ that should be the absolute minimum version.

--

I downloaded FF46 (32 bit) in a Windows 7 VM and installed SAML Tracer v1.0. The web extension could be installed and the icon occurs in the menu bar. But when clicking on that icon... nothing happens (as reported bei @RandomGuy26 and @Bigfoot0007).

To reveal any errors, one should open the "Browser Console" which can be opend by pressing <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd>.
That console shows the following error in FF46 when the SAML Tracer menu item is clicked:
> Type error for parameter createData (Property "type" is unsupported by Firefox) for windows.create.

![image](https://user-images.githubusercontent.com/17160647/35167578-55621da2-fd56-11e7-9ba5-5a6954f0745f.png)

I cannot tell why this happens, but I think that in this case the window-API may not implement everything that's requiered (just a guess), although it should.

I tested the same thing for FF 52.2.3 ESR (32 bit). In this case SAML Tracer opens but doesn't trace any request. Again, the browser console reveals the errors. The errors shown there relate to the logic around webRequest-API. So again: My wild guess is that there isn't everything up to date with that API in FF52.

![image](https://user-images.githubusercontent.com/17160647/35167610-75fb1c1c-fd56-11e7-8dbb-904e80031e62.png)

My last try was FF56  (32 bit). This FF-version worked without any issues with SAML Tracer v1.0 \*phew\* :-)
Maybe there are other compatible FF-versions between FF52 and FF56 as well.

--

**tl;dr**

So to sum this up:
If there's no reason that prevents you from upgrading to FF57 (or any newer version in the future) - upgrade to that version and use FF57+ in conjunction with SAML Tracer v1.0+.

If (for whatever reason) you cannot upgrade to FF57:
* try SAML Tracer v1.0 with e.g. FF56
* or: stick with SAML Tracer v0.3.3 or v0.4.0. (SAML Tracer v0.4.0 doesn't show up at addons.mozilla.org? Go get v0.3.3 from [here](https://github.com/UNINETT/SAML-tracer/releases))

**Comment by thijskh** [comment]
Agreed with khlr: there's nothing wrong with SAML-Tracer 0.4 so you can just continue to use it with older browser versions. And use 1.0 and up for post-57 FF.

**Comment by jaimeperez** [comment]
I think @khlr summarized pretty well what versions of the SAML tracer will work with what versions of Firefox, so I'm closing the issue.

**Comment by ElijahLynn** [comment]
The known bug/issue with a whitescreen unless you right-click should be linked to from this issue.

**Comment by thijskh** [comment]
It's https://bugzilla.mozilla.org/show_bug.cgi?id=1425829
and perhaps https://bugzilla.mozilla.org/show_bug.cgi?id=1408446

**Comment by thijskh** [comment]
It does not happen anymore for me on FF 63
